### DISCUSSION TITLE
Possible to get a texture (as jpeg) of a reconstructed 3D model?

### DISCUSSION THREAD

**Discussion by przemekblasiak** [isl-org/Open3D#4947] [Q&A]
Right now, I reconstruct 3D scenes based on RGBD information and save the resulting mesh as a .gltf model. It's colored (triangles have colors) but I'd like to have that color information as a separate texture file (e.g. jpeg).

I'd appreciate any clues on how to approach it and if it's possible to accomplish with Open3D or other tools.

**Comment by przemekblasiak** [comment]
- There are libraries like [openMVS](https://github.com/cdcseacave/openMVS) that solve the problem of mesh texturing.
- There are papers like [Texture Mapping for 3D Reconstruction with RGB-D Sensor](https://openaccess.thecvf.com/content_cvpr_2018/papers/Fu_Texture_Mapping_for_CVPR_2018_paper.pdf) on the subject.

All in all, I believe it makes sense for Open3D to offer even a simple texture projection algorithm based on RGBD (+odometry) to make the reconstruction pipeline complete.

**Reply by flolu** [reply]
Yes, that's really the last piece missing

**Comment by benjaminum** [comment]
Have you checked `open3d.t.geometry.TriangleMesh`? It has functions for computing uv coordinates and baking vertex and triangle attributes to textures

**Reply by flolu** [reply]
You are right, but it doesn't support projecting textures from multiple camera views onto the mesh
